firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/model/CrashlyticsReport.java

```diff
@@ -0,0 +1,854 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.crashlytics.internal.model;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import com.google.auto.value.AutoValue;
+import com.google.firebase.crashlytics.internal.model.CrashlyticsReport.Session.Application.Organization;
+import com.google.firebase.crashlytics.internal.model.CrashlyticsReport.Session.Event;
+import com.google.firebase.crashlytics.internal.model.CrashlyticsReport.Session.Event.Application;
+import com.google.firebase.crashlytics.internal.model.CrashlyticsReport.Session.Event.Application.Execution.Thread.Frame;
+import com.google.firebase.encoders.annotations.Encodable;
+import com.google.firebase.encoders.annotations.Encodable.Field;
+import com.google.firebase.encoders.annotations.Encodable.Ignore;
+import java.nio.charset.Charset;
+import java.util.List;
+
+@Encodable
+@AutoValue
+public abstract class CrashlyticsReport {
+
+  private static final Charset UTF_8 = Charset.forName("UTF-8");
+
+  @NonNull
+  public static Builder builder() {
+    return new AutoValue_CrashlyticsReport.Builder();
+  }
+
+  @NonNull
+  public abstract String getSdkVersion();
+
+  @NonNull
+  public abstract String getGmpAppId();
+
+  public abstract int getPlatform();
+
+  @NonNull
+  public abstract String getInstallationUuid();
+
+  @NonNull
+  public abstract String getBuildVersion();
+
+  @NonNull
+  public abstract String getDisplayVersion();
+
+  @NonNull
+  public abstract Session getSession();
+
+  @NonNull
+  public CrashlyticsReport withEvents(@NonNull List<Event> events) {
```

Comment:
It's kinda confusing how this lets you build reports but not using the builder itself. Why wouldn't callers just use the builder directly?

Reply:
This allows you to augment an existing report object with a list of events while maintaining all of the other values already set on it. https://github.com/google/auto/blob/master/value/userguide/builders-howto.md#withers

Reply:
I'll add comments to these `with` methods to clarify that.